src/config.js

```diff
@@ -256,7 +259,8 @@ export default class Config {
     this.modulesFolder = opts.modulesFolder;
     this.globalFolder = opts.globalFolder || constants.GLOBAL_MODULE_DIRECTORY;
     this.linkFolder = opts.linkFolder || constants.LINK_REGISTRY_DIRECTORY;
-    this.production = !!opts.production;
+    this.noProduction = opts.production === 'false' || opts.production === '0';
```

Comment:
That is a bit confusing, why do we need both noProduction and production in the class state?